File: MobileTracker/app/src/main/java/me/myds/g2u/mobiletracker/db/BlockDAO.java
package me.myds.g2u.mobiletracker.db;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface BlockDAO {

    @Query("SELECT * FROM BlockEntity ORDER BY time_stamp DESC")
    List<BlockEntity> list();

    @Query("SELECT * FROM BlockEntity WHERE block_hash IN (:hashes)")
    List<BlockEntity> list(String ...hashes);

    @Insert
    void insert(BlockEntity ...blockEntities);

    @Delete
    void delete(BlockEntity ...blockEntities);
}

File: app/src/main/java/io/yena/mobiletracker/db/BlockDao.kt
package io.yena.mobiletracker.db

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*

@Dao
interface BlockDao {

    @Query("SELECT * FROM block")
    fun getAll(): LiveData<List<Block>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(block: Block)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(blocks: List<Block>)

    @Query("SELECT * FROM block WHERE block_hash IS :hash")
    fun getByHash(hash: String): Block

    @Query("SELECT block_hash FROM block")
    fun getAllBlockHashes(): List<String>
}